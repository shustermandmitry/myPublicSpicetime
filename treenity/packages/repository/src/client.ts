export * from './client/access-token';
export * from './client/create-feathers-client';
export * from './client/simple-client';
export * from './utils/cookies';
export * from './client/socket-custom-methods';
export * from './mods/client';
export * from './client/parse-jwt';
export * from './utils/email-parse';
export { makeServiceActions } from '@/mods/entity/service-entity-actions';
