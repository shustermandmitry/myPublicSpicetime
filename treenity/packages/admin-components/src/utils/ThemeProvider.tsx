/*
 * Copyright (c) 2024. Treenity Inc.
 */

import { setCookie, useCookie } from '@treenity/repository/client';
import { TreeNode, useTree } from '@treenity/tree-api';
import { useStorage } from '@treenity/ui-kit/hooks';
import { Theme, ThemeEntity, ThemeProvider } from '@treenity/ui-kit/theme';
import React, { FC, PropsWithChildren } from 'react';
import useSWR from 'swr';

const defaultThemeName = 'light';

function useLoadTheme(name: string): Theme | undefined {
  const treeApi = useTree();
  const { data: themeNode } = treeApi.useNode(`/sys/themes/${name}`);
  const { data } = useSWR(`${name}-theme` + (themeNode as any)?.$id, () =>
    (themeNode as TreeNode)?.get(ThemeEntity),
  );
  return data;
}

const ThemeProviderLoader: FC<PropsWithChildren> = ({ children }) => {
  const lightTheme = useLoadTheme('light');
  const darkTheme = useLoadTheme('dark');

  const cookies = useCookie();
  const [storedThemeName, _setThemeName] = useStorage<string>(
    'theme',
    cookies.theme || defaultThemeName,
  );
  const setThemeName = (theme: string) => {
    setCookie('theme', theme, 365 * 24 * 60 * 60);
    _setThemeName(theme);
  };

  if (!lightTheme || !darkTheme) return null;

  return (
    <ThemeProvider
      light={lightTheme}
      dark={darkTheme}
      children={children}
      storedThemeName={storedThemeName}
      setThemeName={setThemeName}
    />
  );
};

export default ThemeProviderLoader;
