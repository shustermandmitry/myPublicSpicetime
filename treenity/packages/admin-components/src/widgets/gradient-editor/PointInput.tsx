import Button from '@/components/button';
import ColorPicker from '@/components/color-picker';
import Icon from '@/components/icon';
import Input from '@/components/input';
import styled from '@emotion/styled';
import { omitProps } from '@treenity/ui-kit/utils';

import React, { FC } from 'react';
import { IPoint } from './index';

interface PointInputsProps {
  index: number;
  count: number;
  value?: IPoint;
  onChange(values: IPoint): void;
  onRemove(key: number): void;
}

const PointInput: FC<PointInputsProps> = ({ index, count, value, onChange, onRemove }) => {
  const handleChange = <K extends keyof IPoint>(name: K, newValue: IPoint[K]) => {
    const updateValue = {
      ...value,
      [name]: newValue,
    } as IPoint;

    onChange?.(updateValue);
  };

  const isShowButton = count > 2;

  return (
    <PointItem showButton={isShowButton}>
      <Input
        min={0}
        max={100}
        suffix="%"
        type="number"
        size="x-small"
        placeholder="0"
        value={value?.position}
        onChange={e => handleChange('position', Number(e.target.value))}
      />
      <ColorPicker
        format="rgb"
        size="small"
        value={value?.color}
        onChangeComplete={value => handleChange('color', value ? value?.toRgbString() : null)}
      />
      {isShowButton && (
        <Button
          type="secondary-outlined"
          onClick={() => onRemove(index)}
          size="x-small"
          icon={<Icon name="border-solid_outlined" />}
        />
      )}
    </PointItem>
  );
};

export const PointItem = styled('div', omitProps('showButton'))<{ showButton: boolean }>`
  display: grid;
  grid-template-columns: 23% auto ${p => p.showButton && '20px'};
  gap: 4px;

  button {
    i {
      color: ${p => p.theme.colorText};
    }
  }
`;

export default PointInput;
