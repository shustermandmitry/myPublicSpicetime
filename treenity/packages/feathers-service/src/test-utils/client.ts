import {
  Application,
  defaultEventMap,
  defaultServiceMethods,
  RealTimeConnection,
  TransportConnection,
} from '@feathersjs/feathers';
import { Service, SocketService } from '@feathersjs/transport-commons/client';
import { runMethod } from '@feathersjs/transport-commons/lib/socket/utils';

export { SocketService };

export default function testClient<Services = any>(options?: any) {
  const defaultService = function (this: Application<any>, name: string) {
    const self = this;
    const events = Object.values(defaultEventMap);
    const connection = {
      emit(method: string, ...args: any[]) {
        return runMethod(self, {} as RealTimeConnection, name, method, args);
      },
    };

    const settings = Object.assign({}, options, {
      events,
      name,
      connection,
      method: 'emit',
    });

    return new Service(settings) as any;
  };

  const initialize = function (app: Application<Services>) {
    app.defaultService = defaultService;
    app.mixins.unshift((service, _location, options) => {
      if (options && options.methods && service instanceof Service) {
        const customMethods = options.methods.filter(name => !defaultServiceMethods.includes(name));

        service.methods(...customMethods);
      }
    });
  };

  initialize.Service = Service;
  initialize.service = defaultService;

  return initialize as TransportConnection<Services>;
}

if (typeof module !== 'undefined') {
  module.exports = Object.assign(testClient, module.exports);
}
