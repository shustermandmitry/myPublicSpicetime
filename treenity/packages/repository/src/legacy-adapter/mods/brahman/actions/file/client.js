import './TelegramActionFileSend.client';
import './TelegramActionFileUpload.client';
import './TelegramFile.client';
