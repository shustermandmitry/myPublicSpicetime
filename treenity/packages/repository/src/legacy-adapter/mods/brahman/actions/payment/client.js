import './TelegramActionGetWallet.client';
import './TelegramActionSendCoins.client';
import './TelegramActionSplitCoins.client';
