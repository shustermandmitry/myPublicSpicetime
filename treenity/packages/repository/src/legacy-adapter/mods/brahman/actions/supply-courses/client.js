import './TelegramActionSupplyCourses.client';
import './TelegramCoursesOperator.client';
import './TelegramActionBecomeOperator.client';
