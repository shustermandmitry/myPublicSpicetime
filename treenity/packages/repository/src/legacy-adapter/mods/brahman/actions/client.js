import './addActionMenu';
import './back/TelegramActionBack.client';
import './broadcast/TelegramActionBroadcast.client';

import './call-command-by-param/TGActionCallCommandByParam.client';
import './crypto/TelegramActionGetRates.client';
// import './voice/video/SynhesizeAudioVideo';
import './emit-text/EmitText.client';
import './file/TelegramActionFileSend.client';
import './file/TelegramActionFileUpload.client';
import './file/TelegramFile.client';
import './forward/TelegramActionForward.client';
import './game/CurrentGame.client';
import './game/Game.client';
import './game/TelegramActionGame.client';
import './get-value/GetValue.client';
import './i18n/TGActionSelectLanguage.client';
import './if-else/TelegramActionIfElse.client';

import './keyword-select/TelegramActionKeywordSelect.client';
import './menu/TelegramActionMenu.client';
import './message/TelegramActionMessage.client';
import './on-error/TelegramActionOnError.client';
import './page/TelegramActionPage.client';
// import './pantaro/client';
import './params/TelegramActionParams.client';
import './payment/client';
import './question/TelegramActionQuestion.client';
import './removeMessage/TelegramActionRemove.client';
import './reset-history/ResetSession.client';
import './reset-session/ResetSession.client';
import './set-telegram-user/TGActionSetTelegramUser.client';
import './set-value/SetValue.client';

import './supply-courses/client';

import './tag/SetTag';
import './TelegramAction.client';
import './timers/TgActionSetTimer';

import './timers/timers.service.client';

import './translate/TString.client';
import './voice/tinkoff/TinkoffVoiceRecognition.client';
import './voice/tinkoff/TinkoffVoiceSynthesize.client';

import './voice/yandex/YandexVoiceRecognition.client';
import './voice/yandex/YandexVoiceSynthesize.client';

import './ym-statistics/TelegramPageStatistics.client';
