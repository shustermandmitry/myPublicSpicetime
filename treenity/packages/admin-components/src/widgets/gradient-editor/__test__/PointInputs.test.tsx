/*
 * Copyright (c) 2024. Treenity Inc.
 */

import ThemeProvider from '@/utils/ThemeProvider';
import { jest } from '@jest/globals';
import { fireEvent, render } from '@testing-library/react';
import { createCookies, parseCookies, setCookie, CookieContext } from '@treenity/repository/client';
import { IPoint } from '../index';
import PointInput from '../PointInput';

const defaultValue: IPoint = {
  position: 30,
  color: 'rgb(255, 0, 0)',
};

const cookies = createCookies(parseCookies(document.cookie), setCookie);
let valueMock: IPoint = defaultValue;

const updateMock = jest.fn((value: IPoint) => {
  valueMock = value;
  return value;
});

const removeMock = jest.fn((key: number) => key);

describe('PointInputs', () => {
  beforeEach(() => {
    updateMock.mockClear();
    valueMock = defaultValue;
  });

  it('render PointInput component', async () => {
    const { container } = render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <PointInput
            index={0}
            count={2}
            value={valueMock}
            onChange={updateMock}
            onRemove={removeMock}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`div[class="css-1pg4spb"]`);
    const positionInput = root!.querySelector(`input[max="100"][min="0"][type="number"]`);
    const colorTrigger = root!.querySelector(`div[class^="ant-color-picker-trigger"]`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(positionInput).toBeInTheDocument();
    expect(colorTrigger).toBeInTheDocument();
    expect(colorTrigger).toHaveTextContent(defaultValue.color);
  });

  it('change values', async () => {
    const newColor = [255, 255, 0, 100];
    const newPosition = 10;

    const { container } = render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <PointInput
            index={0}
            count={2}
            value={valueMock}
            onChange={updateMock}
            onRemove={removeMock}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`div[class="css-1pg4spb"]`);
    const positionInput = root!.querySelector(`input[max="100"][min="0"][type="number"]`);
    const colorTrigger = root!.querySelector(`div[class^="ant-color-picker-trigger"]`);

    // expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(positionInput).toBeInTheDocument();
    expect(colorTrigger).toBeInTheDocument();
    expect(colorTrigger).toHaveTextContent(defaultValue.color);

    expect(updateMock.mock.calls).toHaveLength(0);

    fireEvent.click(colorTrigger!);

    expect(updateMock.mock.calls).toHaveLength(0);

    const colorPickerPopover = root!.querySelector(`div[class^="ant-popover ant-zoom-big-appear"]`);
    const colorPickerInputWrapper = colorPickerPopover!.querySelector(
      `div[class="ant-color-picker-input-container"]`,
    );
    const colorPickerInputs = colorPickerInputWrapper!.querySelectorAll(
      `input[class="ant-input-number-input"]`,
    );

    expect(colorPickerPopover).toBeInTheDocument();
    expect(colorPickerInputWrapper).toBeInTheDocument();

    expect(colorPickerInputs).toHaveLength(4);

    [...colorPickerInputs].forEach((input, index) => {
      expect(input).toBeInTheDocument();

      if (index === 3) {
        expect(input).toHaveAttribute('aria-valuemax', '100');
        expect(input).toHaveAttribute('aria-valuemin', '0');
      } else {
        expect(input).toHaveAttribute('aria-valuemax', '255');
        expect(input).toHaveAttribute('aria-valuemin', '0');
      }

      fireEvent.change(input!, { target: { value: newColor[index] } });
    });

    expect(updateMock.mock.calls).toHaveLength(1);

    expect(updateMock.mock.calls.at(-1)).toEqual([
      {
        color: `rgb(${newColor[0]}, ${newColor[1]}, ${newColor[2]})`,
        position: defaultValue.position,
      },
    ]);

    fireEvent.change(positionInput!, { target: { value: newPosition } });

    expect(updateMock.mock.calls).toHaveLength(2);

    expect(updateMock.mock.calls.at(-1)).toEqual([
      {
        color: `rgb(${newColor[0]}, ${newColor[1]}, ${newColor[2]})`,
        position: newPosition,
      },
    ]);
  });
});
