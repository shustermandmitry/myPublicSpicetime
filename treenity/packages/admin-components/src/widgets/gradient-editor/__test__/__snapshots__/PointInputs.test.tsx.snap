// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointInputs render PointInput component 1`] = `
.emotion-0 {
  display: grid;
  grid-template-columns: 23% auto;
  gap: 4px;
}

.emotion-0 button i {
  color: rgba(38, 35, 63, 0.88);
}

.emotion-2 {
  height: 20px;
  font-size: 10px;
  border-radius: 4px;
  padding-inline: 6px;
}

.emotion-2.emotion-2 .ant-input-prefix {
  -webkit-margin-end: 6px;
  margin-inline-end: 6px;
}

.emotion-2.emotion-2 i {
  font-size: 12px!important;
}

.emotion-2[type='number']::-webkit-inner-spin-button,
.emotion-2 input[type='number']::-webkit-inner-spin-button {
  display: none;
}

.emotion-2.ant-input {
  font-weight: 500!important;
}

.emotion-2 .ant-input-prefix i {
  color: #27ae60;
}

.emotion-2 .ant-input-prefix {
  -webkit-margin-end: 8px;
  margin-inline-end: 8px;
}

.emotion-2 i {
  color: #d9d9d9;
  -webkit-transition: color 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: color 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

.emotion-2.ant-input-affix-wrapper-focused i {
  color: #27ae60;
}

.emotion-2.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper,
.emotion-2.ant-input-status-error:not(.ant-input-disabled) {
  background-color: #eb57570a;
  border-width: 1px;
}

.emotion-2.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper i,
.emotion-2.ant-input-status-error:not(.ant-input-disabled) i {
  color: #eb5757;
}

.emotion-2.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper i:before,
.emotion-2.ant-input-status-error:not(.ant-input-disabled) i:before {
  content: '\\e91c';
}

.emotion-2.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input,
.emotion-2.ant-input-status-error:not(.ant-input-disabled) input {
  background: transparent;
}

.emotion-2.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::-webkit-input-placeholder {
  color: #eb5757;
}

.emotion-2.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::-moz-placeholder {
  color: #eb5757;
}

.emotion-2.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input:-ms-input-placeholder {
  color: #eb5757;
}

.emotion-2.ant-input-status-error:not(.ant-input-disabled) input::-webkit-input-placeholder {
  color: #eb5757;
}

.emotion-2.ant-input-status-error:not(.ant-input-disabled) input::-moz-placeholder {
  color: #eb5757;
}

.emotion-2.ant-input-status-error:not(.ant-input-disabled) input:-ms-input-placeholder {
  color: #eb5757;
}

.emotion-2.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input,
.emotion-2.ant-input-status-error:not(.ant-input-disabled) input,
.emotion-2.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::placeholder,
.emotion-2.ant-input-status-error:not(.ant-input-disabled) input::placeholder {
  color: #eb5757;
}

.emotion-4 {
  position: relative;
  line-height: 1px;
}

.emotion-4.emotion-4.emotion-4 .ant-collapse-header {
  display: none;
}

.emotion-4.emotion-4.emotion-4 .ant-color-picker-color-block.ant-color-picker-presets-color:before {
  border-radius: 6px;
}

.emotion-4 .ant-color-picker .ant-color-picker-inner .ant-color-picker-presets .ant-collapse-item>.ant-collapse-content>.ant-collapse-content-box {
  padding: 0;
}

.emotion-5 {
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding-left: 6px;
  background-color: #fff;
  box-shadow: none!important;
  height: 20px;
  border-radius: 4px;
}

.emotion-5 .ant-color-picker-trigger-text {
  font-size: 10px;
}

.emotion-5 .ant-color-picker-color-block {
  width: 12px!important;
  height: 12px!important;
}

.emotion-8 {
  box-shadow: none;
  line-height: 100%;
  letter-spacing: -0.04em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-pack: none;
  -ms-flex-pack: none;
  -webkit-justify-content: none;
  justify-content: none;
  color: #27ae60;
  background: transparent!important;
  position: absolute;
  top: calc(50% - 10px);
  right: 0;
  width: 20px;
  height: 20px;
  padding: 2px;
  border: 0;
  color: rgba(38, 35, 63, 0.88);
}

.emotion-8>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-8.ant-btn.ant-btn-round {
  padding-inline: 12px;
}

.emotion-8 i {
  font-size: 16px;
}

.emotion-8>span {
  gap: 6px;
}

.emotion-8 span {
  font-size: 12px;
}

.emotion-8.emotion-8>span>span>i {
  font-size: 16px;
}

.emotion-8:hover {
  color: #ababab!important;
}

.emotion-8:active {
  color: !important;
}

.emotion-8:hover {
  color: #27ae60!important;
}

.emotion-8.emotion-8 i {
  font-size: 12px;
}

.emotion-9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
}

.emotion-10 [class^='icon-'],
.emotion-10 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-11 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

<div
  class="emotion-0"
>
  <span
    class="ant-input-affix-wrapper ant-input-affix-wrapper-sm emotion-1 ant-input-outlined emotion-2"
  >
    <input
      class="ant-input ant-input-sm emotion-1"
      max="100"
      min="0"
      placeholder="0"
      type="number"
      value="30"
    />
    <span
      class="ant-input-suffix"
    >
      %
    </span>
  </span>
  <div
    class="emotion-4"
  >
    <div
      class="ant-color-picker-trigger ant-color-picker-sm emotion-5 emotion-1"
    >
      <div
        class="ant-color-picker-color-block"
      >
        <div
          class="ant-color-picker-color-block-inner"
          style="background: rgb(255, 0, 0);"
        />
      </div>
      <div
        class="ant-color-picker-trigger-text"
      >
        rgb(255, 0, 0)
      </div>
    </div>
    <button
      class="ant-btn emotion-1 ant-btn-text emotion-8"
      type="button"
    >
      <span>
        <span
          class="emotion-9"
        >
          <span
            class="anticon undefined emotion-10"
          >
            <i
              class="icon-close_outlined emotion-11"
            />
          </span>
        </span>
      </span>
    </button>
  </div>
</div>
`;
